fmn/rules/utils.py

```diff
@@ -67,6 +72,30 @@ def get_fas(config):
     return _FAS
 
 
+def _paginate_pagure_data(url, params, key):
+    # Set up the first page query
+    params['page'] = 1
+    next_page_url = url + "?" + urlencode(params)
+
+    while next_page_url is not None:
+        try:
+            response = requests.get(next_page_url, timeout=15)
```

Comment:
With Pagure being inconsistent with speeds, you may want to bump up the timeout.